out_mongo_test.go

```
@@ -0,0 +1,302 @@
+					Env: []string{
+						fmt.Sprintf("MONGO_INITDB_ROOT_USERNAME=%s", mongoUser),
+						fmt.Sprintf("MONGO_INITDB_ROOT_PASSWORD=%s", mongoPassword),
+					},
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				mongoDB = resource
+
+				fluentBitRunOptions.Links = append(fluentBitRunOptions.Links, fmt.Sprintf("%s:%s", resource.Container.ID, resource.Container.Config.Hostname))
+			})
+		})
+
+		AfterEach(func() {
+			if mongoDB == nil {
+				return
+			}
+
+			if _, ok := os.LookupEnv("DEBUG"); ok && CurrentGinkgoTestDescription().Failed {
+				By("Reading mongodb logs", func() {
+					fmt.Fprintln(GinkgoWriter, "==== MongoDB logs ====")
+
+					err := dockerPool.Client.Logs(docker.LogsOptions{
+						Container:    mongoDB.Container.ID,
+						OutputStream: GinkgoWriter,
+						ErrorStream:  GinkgoWriter,
+						Stdout:       true,
+						Stderr:       true,
+						Timestamps:   false,
+						Follow:       false,
+					})
+					defer Expect(err).ToNot(HaveOccurred())
+
+					fmt.Fprintln(GinkgoWriter, "====  Ends logs   ====")
+				})
+			}
+
+			Expect(mongoDB.Close()).To(Succeed())
+		})
+
+		It("Should work", func() {
+			By("Checking entry count", func() {
+				Eventually(func() int {
+					response, err := http.Get(fmt.Sprintf("%s/api/v1/metrics", fluentBitURL))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).To(HaveHTTPStatus(http.StatusOK))
+
+					body, err := ioutil.ReadAll(response.Body)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(response.Body.Close()).To(Succeed())
+
+					var result FluentBitMetrics
+
+					Expect(json.Unmarshal(body, &result)).To(Succeed())
+
+					outputMetrics, ok := result.Output[outputName]
+					Expect(ok).To(BeTrue())
+
+					fmt.Fprintf(GinkgoWriter, "fluent-bit metrics: %s: %+v\n", outputName, outputMetrics)
+
+					processed, ok := outputMetrics["proc_records"]
+					Expect(ok).To(BeTrue(), "proc_records not found for output %s", outputName)
+
+					retried := outputMetrics["retried_records"]
```

No needed anymore?

---

Needed for fail fast